lib/Command.pm

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+
+package Command;
+
+sub new {
+    my $class = shift;
+    bless {}, $class;
+}
+
+sub execute {
+     die "This method must be overridden by a subclass of __PACKAGE__";
+ }
+
+sub validate {
```

Comment:
-1 I think that we don't have to validate the command params.
The only validation that must be performed is the command first letter.